internal/config/credentials.go

```diff
@@ -0,0 +1,131 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/viper"
+)
+
+type CredentialsConfig map[string]*CredentialConfig
+
+func (c *CredentialsConfig) validate() error {
+	for name, v := range *c {
+		if err := v.validate(); err != nil {
+			return fmt.Errorf("credential %q: %w", name, err)
+		}
+	}
+
+	return nil
+}
+
+func (c *CredentialsConfig) setDefaults(v *viper.Viper) error {
```

Comment:
would you mind updating the `cue` and `jsonschema` for config as well? so we can try to keep that in sync